### PR TITLE
sorbet: Comment more files that can't be `strict` because of `undef`

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

- Part of https://github.com/Homebrew/brew/issues/17297.
- Found with `grep -rL "# typed: strict" Library/Homebrew | xargs grep -l "undef "`.
- This stops people from trying to bump them and getting an error that they can't fix because  [it's a Sorbet limitation](https://sorbet.org/docs/error-reference#3008), wasting contributor time.

### CODE REVIEW

**Comment by carlocab** [comment]
I suspect many (if not all) of these `undef`s can be removed, so it may still be possible to make these `typed: strict`

**Comment by issyl0** [comment]
Oooh, in that case, sounds like we should write another issue to investigate removing them?

**Comment by carlocab** [comment]
> Oooh, in that case, sounds like we should write another issue to investigate removing them?

Yep, fine to merge if we have an issue like that.

**Review by carlocab — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Would be nice if there was a wait for a response to "Are all of the undefs we use really necessary?" before we went "yes, let's remove a bunch" 😉

**Comment by issyl0** [comment]
This PR doesn't remove a bunch, it just notes that we can't Sorbet strict these files.

The issue doesn't remove any either, just lays the groundwork with where they exist, and invites someone to answer the question.

I don't think at any point we said "yes let's remove them"?

**Comment by MikeMcQuaid** [comment]
> This PR doesn't remove a bunch, it just notes that we can't Sorbet strict these files.

Doh, I'm sorry 🤦🏻. Misread. Idiot, apologies!

**Pull request merged**
